src/main/java/org/wildfly/naming/client/SimpleName.java

```diff
@@ -128,9 +128,7 @@ public SimpleName(final Enumeration<String> enumeration) {
                 b.appendCodePoint(ch);
             }
         }
-        if (b.length() > 0 || ch == '/') {
-            segments.add(b.toString());
-        }
+        segments.add(b.toString());
```

Comment:
IMO we should always add the segment to the list.
F.e. if we are parsing ejb address and app name is empty then the first part is parsed as "ejb:" - the name is an empty string. But we have to add it to the list because it is the first part of the address, without that after performing of addAll operation during second step of lookup we will obtain incomplete address. Another possibility is address finishing with "/" but it seems to my that it also should be parsed as address with the last part empty.